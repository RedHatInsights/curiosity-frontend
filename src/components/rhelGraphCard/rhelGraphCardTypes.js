import { translate } from '../i18n/i18n';
import { RHSM_API_QUERY_GRANULARITY_TYPES as GRANULARITY_TYPES } from '../../types/rhsmApiTypes';

const getGranularityOptionsType = optionsType => {
  if (optionsType === 'default') {
    return [
      { title: translate('curiosity-graph.dropdownDaily'), value: GRANULARITY_TYPES.DAILY },
      { title: translate('curiosity-graph.dropdownWeekly'), value: GRANULARITY_TYPES.WEEKLY },
      { title: translate('curiosity-graph.dropdownMonthly'), value: GRANULARITY_TYPES.MONTHLY },
      { title: translate('curiosity-graph.dropdownQuarterly'), value: GRANULARITY_TYPES.QUARTERLY }
    ];
  }

  return [];
};

const rhelGraphCardTypes = {
  getGranularityOptions: getGranularityOptionsType
};

export { rhelGraphCardTypes as default, rhelGraphCardTypes, getGranularityOptionsType };
