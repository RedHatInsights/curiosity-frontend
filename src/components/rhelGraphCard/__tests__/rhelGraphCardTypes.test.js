import { rhelGraphCardTypes, getDateMenuOptionsType } from '../rhelGraphCardTypes';

describe('RhelGraphCardTypes', () => {
  it('should have specific types', () => {
    expect(rhelGraphCardTypes).toMatchSnapshot('rhelGraphCardTypes');
  });

  it('should return a specific output for getDateMenuOptionsType', () => {
    expect(getDateMenuOptionsType()).toMatchSnapshot('getDateMenuOptionsType');
  });
});
