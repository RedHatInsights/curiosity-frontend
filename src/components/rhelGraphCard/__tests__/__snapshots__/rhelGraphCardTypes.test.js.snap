// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RhelGraphCardTypes should have specific types: rhelGraphCardTypes 1`] = `
Object {
  "getGranularityOptions": [Function],
}
`;

exports[`RhelGraphCardTypes should return an output for granularity selection: getGranularityOptionsType 1`] = `Array []`;

exports[`RhelGraphCardTypes should return an output for granularity selection: getGranularityOptionsType:default 1`] = `
Array [
  Object {
    "title": "t(curiosity-graph.dropdownDaily)",
    "value": "daily",
  },
  Object {
    "title": "t(curiosity-graph.dropdownWeekly)",
    "value": "weekly",
  },
  Object {
    "title": "t(curiosity-graph.dropdownMonthly)",
    "value": "monthly",
  },
  Object {
    "title": "t(curiosity-graph.dropdownQuarterly)",
    "value": "quarterly",
  },
]
`;
